Add test for log processing

elkoniu opened the pull request:
'/usr/libexec/anaconda/log-capture' is an util to generate runtime logs. This commit adds test validating output of this util.

M4rtinK commented on efi-log.ks.in:
```diff
@@ -0,0 +1,36 @@
+%ksappend repos/default.ks
+network --bootproto=dhcp
+
+bootloader --timeout=1
```
I think other than the %post script this kickstart tests does not really need anything special, right ? In that case I would suggest using of the "common boilerplate" kickstart fragments instead to inject the common things. For example this one:

https://github.com/rhinstaller/kickstart-tests/blob/master/fragments/shared/common/common_no_payload.ks

A similar (mainly needs to do stuff in %post) kickstart tests that uses this fragment:

https://github.com/rhinstaller/kickstart-tests/blob/2ca0796cd976eb7561fef361f0830005b2196398/ui_cmdline.ks.in#L6

elkoniu replied:
Updated according to suggestion.

M4rtinK commented on efi-log.ks.in:
```diff
@@ -0,0 +1,36 @@
+
+%end
+
+%post
```
The validation boilerplate could be replaced by this fragment: https://github.com/rhinstaller/kickstart-tests/blob/master/fragments/shared/validation/success_if_result_empty_standalone.ks

elkoniu replied:
Updated according to suggestion

rvykydal replied:
Inside `%post` we should use non-`_standalone` version of the fragment.

rvykydal replied:
Or just call the `_standalone` version without %post (standalone :))

elkoniu replied:
Thanks for pointing this our - fixed now.

M4rtinK reviewed (approved):
Looks good to me - thanks! :)

Just added some suggestions for using fragments to provide generic/boilerplate bits in the kickstart via kickstart fragments. :)

elkoniu commented:
@M4rtinK thanks for the review. I have updated the code following your guidelines. Also I have split test cases in half: check if the util exists, call the util and expect `0` return code. Do you thing this is good enough validation?

KKoukiou commented:
/test-tmt

KKoukiou commented:
/test-os-variants

M4rtinK commented:
> @M4rtinK thanks for the review. I have updated the code following your guidelines. Also I have split test cases in half: check if the util exists, call the util and expect `0` return code. Do you thing this is good enough validation?

Agreed with the split - like this we can easily see from the logs what exactly went wrong, nice! :)

As for validation - it creates a tarball, right ? I guess we could check the file exists and is of non-zero size ? But I leave it up to you, given we already check the return code. :)

elkoniu commented:
@M4rtinK @KKoukiou I think this PR is in final shape now and it is passing when I trigger it locally.

rvykydal commented on efi-log.sh:
```diff
@@ -0,0 +1,24 @@
+
+# Ignore unused variable parsed out by tooling scripts as test tags metadata
+# shellcheck disable=SC2034
+TESTTYPE="efi log coverage"
```
- I think the test does not belong to coverage (it is ~30 selected tests).
- Should it be rather `logs` than `log` (we already have tests in the former category)
- Is it of `efi` type? I don't see it runs on uefi in the tests (like for example in https://github.com/rhinstaller/kickstart-tests/blob/54ab7e2db056141791784712907ff467c7b55a3e/reboot-uefi.sh#L24). Is it required for the test to run on uefi? If so then i'd rather use `uefi` both as the type and in the name, again we already have 2 similar tests.

elkoniu replied:
Done

rvykydal commented on efi-log-check.sh:
```diff
@@ -0,0 +1,33 @@
+
+# Check if log utils produces log tarbal
+function check_log_util_produces_log_archive() {
+    local exit_code=0
```
I think this can be removed.

elkoniu replied:
Done

rvykydal commented on efi-log-check.sh:
```diff
@@ -0,0 +1,33 @@
+    /usr/libexec/anaconda/log-capture
+
+    if [[ $? -ne ${exit_code} ]]; then
+        echo "*** Failed check: log-capture util cant be called" >> ${RESULT_FILE}
```
I'd maybe rather say it failed and put the exit code value in. But I don't insist.

elkoniu replied:
Added

rvykydal reviewed (changes requested):
Nice work.

We should perhaps clear up the TESTTYPE though.

Also I wouldn't create an include (lib) script at this point where the functions are used only in a single test. But if we create it I'd keep a (silent) convention of having `-lib-` in the name, ideally with `post` and `nochroot` hint. But this can be done updated / discussed in a followup PR. Having more tests using the shared script would also make easier to decide where to put it or how to name it.

elkoniu commented:
Thanks for the review @rvykydal 
I just pushed the latest version and tried to address all of your comments

> Nice work.
> 
> We should perhaps clear up the TESTTYPE though (as suggested above)
> 
> Also I wouldn't create an include (lib) script at this point where the functions are used only in a single test. But if we create it I'd keep a (silent) convention of having `-lib-` in the name, ideally with `post` and `nochroot` hint. But this can be done updated / discussed in a followup PR. Having more tests using the shared script would also make easier to decide where to put the functions or how to name a new file if we create such.

I like this idea. On my todo list there is a small refactoring of the repo (which includes making default repo README more visible). I would like to address this '-lib-' idea as a part of this refactoring.

rvykydal commented:
/test-tmt

rvykydal reviewed (approved):
Looks good to me, thank you.
